gradle-baseline-java/src/main/groovy/com/palantir/baseline/tasks/CheckClassUniquenessLockTask.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.tasks;
+
+import com.google.common.collect.ImmutableList;
+import java.io.File;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.Task;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.ModuleVersionIdentifier;
+import org.gradle.api.provider.SetProperty;
+import org.gradle.api.specs.Spec;
+import org.gradle.api.tasks.CacheableTask;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.util.GFileUtils;
+
+@CacheableTask
+public class CheckClassUniquenessLockTask extends DefaultTask {
+
+    private static final String HEADER = "# Danger! Multiple jars contain identically named classes. This may "
+            + "cause different behaviour depending on classpath ordering.\n"
+            + "# Run ./gradlew checkClassUniqueness --write-locks to update this file\n\n";
+
+    // not marking this as an Input, because we want to re-run if the *contents* of a configuration changes
```

Comment:
shouldn't the contents only change if a dependency on the configuration changes?

Reply:
Configurations are not Serializable, so I can't just mark a list of them as an `@Input`.

Reply:
you can take the configuration as an input by annotating it with @classpath

Reply:
sure, but in order to preserve the ability to analyze [multiple configurations](https://github.com/palantir/gradle-baseline/pull/1145#discussion_r363867958), I don't have just one single configuration here but a bunch of them?

Reply:
Instead of storing a set of configurations could you create a single configuration which extends from all of the provided configurations

Reply:
I dunno, it seems that has a slight risk of false positives... e.g. if we're analyzing two configurations foo and bar and we move a jar from foo -> bar.  In the big clobbered configuration there would be no change, but _actually_ I would expect the lockfile to change.

Reply:
ok no big deal